src/ImageSharp/Formats/Bmp/BmpInfoHeader.cs

```diff
@@ -91,5 +122,40 @@ internal sealed class BmpInfoHeader
         /// or 0 when every color is important{ get; set; } generally ignored.
         /// </summary>
         public int ClrImportant { get; set; }
+
+        /// <summary>
+        /// Parses the full BITMAPINFOHEADER header (40 bytes).
+        /// </summary>
+        /// <param name="data">The data to parse.</param>
+        /// <returns>Parsed header</returns>
+        /// <seealso href="https://msdn.microsoft.com/en-us/library/windows/desktop/dd183376.aspx"/>
+        public static BmpInfoHeader Parse(ReadOnlySpan<byte> data)
+        {
+            return MemoryMarshal.Cast<byte, BmpInfoHeader>(data)[0];
+        }
+
+        /// <summary>
+        /// Parses the BITMAPCOREHEADER consisting of the headerSize, width, height, planes, and bitsPerPixel fields (12 bytes).
+        /// </summary>
+        /// <param name="data">The data to parse,</param>
+        /// <returns>Parsed header</returns>
+        /// <seealso href="https://msdn.microsoft.com/en-us/library/windows/desktop/dd183372.aspx"/>
+        public static BmpInfoHeader ParseCore(ReadOnlySpan<byte> data)
+        {
+            return new BmpInfoHeader(
```

Comment:
Just wondering why don't you do the casting trick here?

Reply:
The core header is only 12 bytes -- so  the cast won't work here. Adding one more check for the full cast case to make sure we're always passing 40 bytes...